### PR TITLE
Add ability to manage proxmox storage.

### PR DESCRIPTION
**Pull request by Fbrachere**
As asked in #35 , a first attempt to manage storages with this role.
For now, the working type storage are:
- nfs
- dir
- rbd

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -27,3 +27,4 @@ pve_ssl_letsencrypt: false
 pve_groups: []
 pve_users: []
 pve_acls: []
+pve_storage: []
```
Comment (lae) [inline defaults/main.yml]: should be `pve_storages` to match the task (test is failing otherwise)

**Comment by lae** [comment]
Can you add some test data in `group_vars/all` and some tests in `tests/test.yml` validating that the storages exist?

File: library/proxmox_storage.py [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    result['state'] = storage.state
+    result['changed'] = False
+
+    if storage.state == 'absent':
+        if storage.exists():
+            result['changed'] = True
+            if module.check_mode:
+                module.exit_json(**result)
+            (changed, error) = storage.remove_storage()
+
```
Comment (lae) [inline library/proxmox_storage.py]: can you remove this newline?

**Comment by lae** [comment]
We can add a real NFS server to the test (see lae.nfs :) ) but I guess it wouldn't be worth it.

Just pored over this PR in more detail and overall it looks good to me + playbook run completes fine. The test tasks have the `pve_storage` typo though, which needs fixing. I'll merge afterwards.

**Comment by lae** [comment]
Actually - I just noticed because of your opening PR comment. If none of the other storage types work, can we just remove them from the choices entirely, or at least comment them out, so no one tries to accidentally use them just because they're *there*?

Also sorry, I just remembered I was going to say this earlier. Can you add a `Storage` section to the documentation (`README.md`) detailing how to setup storages?

**Comment by lae** [comment]
I'm restarting the failed builds, looked like a temporary networking issue or something trying to install dependencies.

**Comment by lae** [comment]
@Fbrachere I'm sure you've noticed by now, but I've given you permissions to commit to this repo, so feel free to push feature branches directly to this repo/open PRs whenever you feel like it. (and if you'd rather not have that ability, let me know.)

**Comment by Fbrachere** [comment]
Ok, thank you for your trust, I'll test this by adding another storage type.

**Pull request merged**